### PR TITLE
feat: add generator for virtual piece loader

### PR DESCRIPTION
**Pull request by MRDGH2821**
Add loader file generator command for virtual pieces of `@sapphire/framework@4.8.2`

PR Inspired by - https://discord.com/channels/737141877803057244/1049419304405311564/1175132383033438218

### CODE REVIEW

**Comment by favna** [comment]
@sapphiredev pack

**Comment by favna** [comment]
Tested with a newly generated project:
ishare-1700405115

File: src/functions/CreateComponentLoader.ts [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+/**
+ * Regular expression pattern for matching files ending with JavaScript-like extensions.
+ * - `.js`
+ * - `.ts`
+ * - `.mjs`
+ * - `.cjs`
+ * - `.mts`
+ * - `.cts`
+ */
+const regexForFilesEndingWithJavaScriptLikeExtensions = /\.(m|c)?(?:j|t)s$/;
```
Comment (kyranet) [inline src/functions/CreateComponentLoader.ts:18]:
```suggestion
const regexForFilesEndingWithJavaScriptLikeExtensions = /\.[mc]?[jt]s$/;
```

**Review by kyranet — CHANGES_REQUESTED** [review]

File: src/functions/CreateComponentLoader.ts [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+ *
+ * We wrap the bulk of this function in a {@link Result.fromAsync} so that if any of the file writing fails
+ * that will bubble up as a failure overall.
+ *
+ * @param targetDir The directory that the `_load.<ext>` file will be written to
+ * @param config The config
+ * @returns Whether the loaders were created successfully
+ */
+export async function CreateComponentLoaders(targetDir: string, config: Config) {
+	const templateHeader = `// import this file in your entry point (index.${config.projectLanguage}) to load respective pieces`;
```
Comment (vladfrangu) [inline src/functions/CreateComponentLoader.ts:55]: Isn't this technically not needed for fs-based projects @kyranet? The file *will* get auto-imported, but nothing will be loaded from it 👀 

Also would be neat if we could make the entry point be taken from package.json but i will not block the PR on that

Comment (kyranet) [reply src/functions/CreateComponentLoader.ts:55]: The FS loader will load pieces twice if they're also loaded with `loadPiece`, first as virtual, then as non-virtual, that's why it should be disabled. That being said, `loadPiece` needs to be called for each piece, as such, this generator isn't fully automated.

The entrypoint bit is adding `import './listeners/_load.ts';` in the entrypoint file so the virtual files are registered, we can't automate this without using the TS API.

**Review by kyranet — APPROVED** [review]

**Pull request merged**